server/src/main/java/org/elasticsearch/common/time/JavaDateMathParser.java

```diff
@@ -46,11 +47,11 @@
  */
 public class JavaDateMathParser implements DateMathParser {
 
-    private final DateTimeFormatter formatter;
+    private final JavaDateFormatter formatter;
     private final DateTimeFormatter roundUpFormatter;
     private final String format;
 
-    JavaDateMathParser(String format, DateTimeFormatter formatter, DateTimeFormatter roundUpFormatter) {
+    JavaDateMathParser(String format, JavaDateFormatter formatter, DateTimeFormatter roundUpFormatter) {
```

Comment:
why this change?

Reply:
the intention was to allow alternatives in patterns for date math calculations. 
We already discussed this and we do not need this. 
I think that in this case I will revert this change and create  with `JavaDateMathParser(firstParser,getRoundUpFormatter)`

Reply:
actually this should stay. There is plenty of tests that started to fail now because of that.
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-1/10768/
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-2/10640/
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-bwc/2575/
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request+oss-distro-docs/13976/